engine/execution/state/state.go

```diff
@@ -410,31 +413,35 @@ func (s *state) saveExecutionResults(
 		return fmt.Errorf("can not retrieve chunk data packs: %w", err)
 	}
 
-	// Acquire both locks to ensure it's concurrent safe when inserting the execution results and chunk data packs.
-	return storage.WithLocks(s.lockManager, []string{storage.LockInsertOwnReceipt, storage.LockInsertChunkDataPack}, func(lctx lockctx.Context) error {
-		err := s.chunkDataPacks.StoreByChunkID(lctx, chunks)
-		if err != nil {
-			return fmt.Errorf("can not store multiple chunk data pack: %w", err)
-		}
-
-		// Save entire execution result (including all chunk data packs) within one batch to minimize
-		// the number of database interactions.
+	storeFunc, err := s.chunkDataPacks.Store(chunks)
```

Comment:
I would suggest to rename `storeFunc` to something more specific, reflecting the functor's conceptual importance. 

```suggestion
	commitment2ChunkDataPacks, err := s.chunkDataPacks.Store(chunks)
```